skopt/plots.py

```diff
@@ -88,3 +89,65 @@ def plot_convergence(*args, **kwargs):
         ax.legend(loc="best")
 
     return ax
+
+
+def plot_scatter_matrix(result, bins=20):
+    """Plot scatter plot matrix of samples.
+
+    The diagonal shows a histogram for each dimension. Create pairwise
+    scatter plots of each dimension of the search space. Later
+    samples are shown in a darker color.
+
+    Parameters
+    ----------
+    * `result` [`OptimizeResult`]
+        The result for which to plot the scatter plot matrix.
+
+    * `bins` [int, bins=20]:
+        Number of bins to use for histograms on the diagonal.
+
+    Returns
+    -------
+    * `ax`: [`Axes`]:
+        The matplotlib axes.
+    """
+    samples = np.asarray(result.x_iters)
+    order = range(samples.shape[0])
+    fig, ax = plt.subplots(result.space.n_dims,
+                           result.space.n_dims,
+                           figsize=(8, 8))
+    fig.subplots_adjust(left=0.05, right=0.95, bottom=0.05, top=0.95,
+                        hspace=0.1, wspace=0.1)
+
+    for i in range(result.space.n_dims):
+        for j in range(i + 1):
+            if i == j:
+                ax[i, i].hist(samples[:, i], bins=bins)
```

Comment:
Then if points correspond to their objective value, a better plot on the diagonal might be the partial dependence plots of those dimensions? what do you think?